spec/support/shared_examples_policy_machine.rb

```diff
@@ -1173,4 +1173,95 @@
 
     end
   end
+
+  describe 'batch_pluck' do
+
+    before do
+      @one_fish = policy_machine.create_object('one:fish')
+      @two_fish = policy_machine.create_object('two:fish')
+      @red_one = policy_machine.create_object('red:one')
+      @read = policy_machine.create_operation('read')
+      @write = policy_machine.create_operation('write')
+      @u1 = policy_machine.create_user('u1')
+      @ua = policy_machine.create_user_attribute('ua')
+      [@one_fish, @two_fish, @red_one].each do |object|
+        policy_machine.add_association(@ua, Set.new([@read]), object)
+      end
+      @oa = policy_machine.create_object_attribute('oa')
+      policy_machine.add_association(@ua, Set.new([@write]), @oa)
+      policy_machine.add_assignment(@u1, @ua)
+      policy_machine.add_assignment(@red_one, @oa)
+    end
+
+    context 'when given a block' do
+
+      it 'calls the block' do
+        expect do |spy|
+          policy_machine.batch_pluck(type: :object, query: { unique_identifier: 'one:fish' }, fields: [:unique_identifier], &spy)
+        end.to yield_control
+      end
+
+      context 'and search terms' do
+        it 'returns the matching attributes' do
```

Comment:
Mind also adding tests about the returned objects' class and that they don't respond to attributes that aren't in the fields array?